provider/provider.go

```diff
@@ -299,3 +357,106 @@ func parseCheckpointObject(obj resource.PropertyMap) resource.PropertyMap {
 
 	return nil
 }
+
+type contextHash struct {
```

Comment:
I understand the code below is borrowed - a link perhaps? That actually is great increases my confidence it's battle tested and less need to test for us.

Reply:
Here's Materialize's implementation: https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/master/cmd/pulumi-resource-docker-buildkit/main.go#L387

Reply:
Update: I found a bug when not `dockerignore`ing the `node_modules` folder.
Expected (desired) behavior: Dockerfile rebuilds on any non-`dockerignore`'d change
Actual behavior: Provider crashes with unexpected error
`unable to hash build context: hashing "node_modules/@pulumi/docker": read node_modules/@pulumi/docker: read node_modules/@pulumi/docker: is a directory`

That's not OK to merge. Will look into it.

Reply:
This should be resolved in the latest commit.